### PR TITLE
mitosheet: create comms manually, instead of relying on widget infrastructure

### PR DESCRIPTION
**Pull request by naterush**
# Description

Implements part 1 of: https://www.notion.so/trymito/Jupyter-Widgets-Removal-Sprint-2-daf1536f8e834f418162efe837140f16

# Testing

Test that it works! Do a workflow!

# Documentation

No.

### CODE REVIEW

File: mitosheet/mitosheet/mito_widget.py [hunk 1]
```diff
@@ -223,6 +237,8 @@ def receive_message(self, widget: Any, content: Dict[str, Any], buffers: Any=Non
         4. A log_event is just an event that should get logged on the backend.
         """
 
+        print("RECEIVED", content)
```
Comment (naterush) [inline mitosheet/mitosheet/mito_widget.py:240]: Remove

File: mitosheet/src/plugin.tsx [hunk 2]
```diff
@@ -84,6 +84,17 @@ function activateWidgetExtension(
     // Add the Create New Mitosheet button
     addButton(tracker);
 
+    app.commands.addCommand('create-mitosheet-comm', {
+        label: 'Create Comm',
+        execute: (args: any) => {
+            const comm_target_id = args.comm_target_id;
+            const currentKernel = tracker.currentWidget?.context.sessionContext?.session?.kernel;
+            const comm = currentKernel?.createComm(comm_target_id);
+            comm?.open()
```
Comment (naterush) [inline mitosheet/src/plugin.tsx:93]: Note this

**Review by naterush — COMMENTED** [review]
Minor cleanup!

**Comment by naterush** [comment]
NOTE: do not merge this. We don't want this in this release!

**Review by aarondr77 — COMMENTED** [review]
I did an analysis of the logs with this and didn't see any difference. LGTM

File: mitosheet/src/jupyter/api.tsx [hunk 3]
```diff
@@ -145,33 +214,47 @@ export const useMitoAPI = (
...
-        registerReceiveHandler(this.receiveResponse.bind(this))
         this.unconsumedResponses = [];
     }
 
+    // NOTE: see comment in useMitoAPI above. We will get rid of this function
+    // when we move away from the widget framework and have a better place to
+    // call async functions
+    async init(commContainer: CommComtainer | undefined): Promise<void> {
+        if (commContainer === undefined) {
+            // TODO: display an error here. Notably,
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/src/jupyter/api.tsx:246]: We unfortunately don't have access to logging, as we don't have an API, as we can't create comms! This is where we'll display an error, coming in next PR.

File: mitosheet/src/jupyter/api.tsx [hunk 4]
```diff
@@ -192,9 +275,22 @@ export default class MitoAPI {
...
 
-        // Send the message
-        this._send(msg);
+        // Send the message. 
+        // NOTE: see comment in useMitoAPI above. We will get rid of this wait if the comms don't
+        // yet exist, once we move away from the widget framework. For now, it might happen that 
+        // we send a message before the comm is created, but this timeout and wait should be
+        // enough to make this not a problem
+        if (this._send === undefined) {
+            console.log("Send not yet defined, waiting");
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/src/jupyter/api.tsx:284]: See above.

**Pull request merged**